authentication/tests/test_user_login.py

```diff
@@ -0,0 +1,62 @@
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from authentication.models import CustomUser
+
+
+class UserLoginAPITests(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.test_user = CustomUser.objects.create_user(
+            person_email="test@test.com",
+            password="Test1234",
+            person_name="Test",
+            person_surname="Test",
+        )
+        cls.test_user.is_active = True
+        cls.test_user.save()
```

Comment:
I strongly recommend to not use `setUpTestData()` (and use `setUp`) instead.

Yes, it will make tests execution significantly longer (especially if we create a lot of stuff in setUp). But there is a lot less possible sideeffects.

Let's say, if someone will update or remove instances created in `setUpTestData` — you'll get flaky tests (sometimes they will pass, sometimes not, depending on order of execution).

I use `setUpTestData` only in cases when there are strong reasons for this. In general, tests are pretty fast, so `setUp` is the recommended appoach (as it is a significantly safer one).

Reply:
setUpTestData() replaced with setUp()
Thanks!